### DISCUSSION TITLE
Spdlog sink for integration with Seq/GELF/Greylog

### DISCUSSION THREAD

**Discussion by sketch34** [gabime/spdlog#2341] [Q&A] [answered]
I'd love to have my spdlog logs appear in [Seq](https://datalust.co/seq). It looks like the way to achieve this is [via GELF](https://docs.datalust.co/docs/using-gelf).

I found this: https://github.com/ess-dmsc/conan-spdlog
And this: https://github.com/gabime/spdlog/issues/673

Any other efforts on this front, what is the latest on this?

**Comment by tt4g** [comment, marked as answer]
Additional attributes are not yet supported.
However, there is an issue #617 that is currently open and [this comment](https://github.com/gabime/spdlog/issues/617#issuecomment-1018751932) introduces [structured_spdlog](https://github.com/bobhansen/structured_spdlog) that additional attribute features.

**Reply by sketch34** [reply]
Okay so attributes are required in order to conform to and utilize Seq's expected GELF (JSON) format. And then we would need a spdlog sink that could send the GELF log data to Seq over a socket connection.

Is this a summary of what we would need?

**Reply by tt4g** [reply]
The additional attributes were discussed in #673 and I thought you might need them, so I introduced them.
As far as the following documentation is concerned, the only required attribute for HTTP endpoints is `@t` (timestamp).

https://docs.datalust.co/docs/posting-raw-events

If you are creating a custom sink, you can implement `spdlog::sinks::base_sink::sink_it_(const spdlog::details::log_msg &msg)` to access log time information.
And log message can then be generated by formatter.
If you are using the HTTP endpoint, it appears that Seq can be used without the additional attributes.
